presto-spi/src/main/java/com/facebook/presto/spi/ConnectorSplit.java

```diff
@@ -48,4 +50,12 @@ default OptionalLong getSplitSizeInBytes()
     {
         return OptionalLong.empty();
     }
+
+    /**
+     * Provide a set of columns whose domains could be ignored.
+     */
+    default Set<ColumnHandle> getRedundantColumnDomains()
+    {
+        return new HashSet<>();
+    }
```

Comment:
The only concern I have is the SPI change. Let's chat offline

Reply:
As discussed, we will pass `ConnectSplit` to `ConnectorTableLayoutHandle#getIdentifier`, so that engine will not bother connector-specific runtime processing.